VPLAY-11267 : SLE thumbnails presented to the user is drifted by time…

Gnanesha opened the pull request:
…s ranging 10-15 seconds

Reason for change : added logic to handle sle thumbnail
Risks: LOW
Test Procedure: test SLE with thumbnails
Priority: P0

Gnanesha committed:
VPLAY-11267 : SLE thumbnails presented to the user is drifted by times ranging 10-15 seconds

Reason for change : added logic to handle sle thumbnail
Risks: LOW
Test Procedure: test SLE with thumbnails
Priority: P0

Signed-off-by: Gnanesha <gnaani82@gmail.com>

Gnanesha committed:
VPLAY-11267 : SLE thumbnails presented to the user is drifted by times ranging 10-15 seconds

Reason for change : added logic to handle sle thumbnail
Risks: LOW
Test Procedure: test SLE with thumbnails
Priority: P0

Signed-off-by: Gnanesha <gnaani82@gmail.com>

rdkcmf-jenkins commented on fragmentcollector_hls.cpp:
```diff
@@ -5372,7 +5452,66 @@ bool StreamAbstractionAAMP_HLS::SetThumbnailTrack( int thumbIndex )
+			}
+			else
+			{
+				AAMPLOG_WARN("not found matching starttime:%lf %lf",findIter->startTime,tStart);
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
8de308c62a94addfcad6a1b550e49d3c
NOT_PRESENT
-->

Coverity issue no longer present as of: undefined
<details>
<summary>Show issue</summary>


# Coverity Issue - Using invalid iterator
Dereferencing iterator "findIter" though it is already past the end of its container.

_Medium Impact, CWE-119_
_INVALIDATE_ITERATOR_



</details>

pstroffolino replied:
can we fix this?

Gnanesha replied:
yes

Gnanesha committed:
VPLAY-11267 : SLE thumbnails presented to the user is drifted by times ranging 10-15 seconds

Reason for change : added logic to handle sle thumbnail
Risks: LOW
Test Procedure: test SLE with thumbnails
Priority: P0

pstroffolino commented on fragmentcollector_hls.cpp:
```diff
@@ -5280,22 +5357,25 @@ std::vector<TileInfo> IndexThumbnails( lstring iter , double stTime=0 )
 			{
 				TileInfo tileInfo;
-				if( 0.0f == layout.posterDuration )
+				if( ptr.tostring().length() > 4 )
```
what is this?  why are we exporting as string, only to measure length?

Gnanesha replied:
Some times Pointer is returning complete string till end of file with no proper link for thumbnail, So when  exported to string it is returning single line. 
2. Even after exporting , no proper image url is returned . sometimes it returns empty string even though string is not empty. this is leading to missing the image url.  but length check is solving the issue. 
3. There is hidden bug introduced when revamp of HLS parsing is done from 8.1 onwards. however it is getting exposed now.  I have done extensive testing and it solves the problem.

pstroffolino committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-11267

Gnanesha committed:
VPLAY-11267 : SLE thumbnails presented to the user is drifted by times ranging 10-15 seconds

Reason for change : added logic to handle sle thumbnail
Risks: LOW
Test Procedure: test SLE with thumbnails
Priority: P0

pstroffolino committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-11267

pstroffolino committed:
Reason for Change: small optimization

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>
